### PR TITLE
{data}[iimpi/2022a] netCDF-Fortran v4.6.0, netCDF v4.9.0

### PR DESCRIPTION
**Pull request by SimonPinches**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/n/netCDF/netCDF-4.9.0-iimpi-2022a.eb
Comment (SebastianAchilles) [inline easybuild/easyconfigs/n/netCDF/netCDF-4.9.0-iimpi-2022a.eb:35]: The style checker doesn't like this. Either remove this line, or indent the comment
```suggestion
```

```
    # ('bzip2', '1.0.8'),
```

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen2

File: easybuild/easyconfigs/n/netCDF/netCDF-4.9.0-iimpi-2022a.eb
Comment (SimonPinches) [reply easybuild/easyconfigs/n/netCDF/netCDF-4.9.0-iimpi-2022a.eb]: It's toast ;-)

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ generoso

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
zen2-ubuntu-eb - Linux Ubuntu 22.04, x86_64, AMD EPYC 7452 32-Core Processor (zen2), Python 3.10.4
See https://gist.github.com/36aa99ff1e3f59dff967d9174d16b501 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
bdw-opensuse-154 - Linux openSUSE Leap 15.4, x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), 2 x NVIDIA NVIDIA GeForce GTX 1060 6GB, 510.85.02, Python 3.6.15
See https://gist.github.com/d496983741572cd3e9a51d535b4d8a9c for a full test report.

**Review by Micket — CHANGES_REQUESTED** [review]
please add the netCDF-%(version)s_skip-nasa-test.patch patch that was just merged.

**Comment by Micket** [comment]
uhm, i don't think you should have merged the commit (since that commit is already merged into develop). This needs to be rebased on develop anyway now

**Comment by SimonPinches** [comment]
@Micket, yeah, I think I got a bit carried away there.  I'll reset to before the merge and then rebase onto develop.

**Review by Micket — APPROVED** [review]
lgtm

**Comment by Micket** [comment]
@boegelbot please test @ generoso

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in total)
alvis-c1 - Linux Rocky Linux 8.5, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/d0db527260df8e0c286abc64d0dc5475 for a full test report.

**Comment by Micket** [comment]
ugh
```
206/227 Test #206: dap4_test_test_thredds ................***Failed  129.72 sec
unknown argument test_thredds.sh
test_thredds.sh:
testing: dap4://thredds-test.unidata.ucar.edu/thredds/dap4/casestudies/harvey/goes16/CONUS/Channel01/20170821/GOES16_CONUS_20170821_020218_0.47_1km_33.3N_91.4W.nc4?dap4.ce=y#checksummode=ignore
Error:curl error: Timeout was reached
/dev/shm/netCDF/4.9.0/iimpi-2022a/easybuild_obj/ncdump/ncdump: dap4://thredds-test.unidata.ucar.edu/thredds/dap4/casestudies/harvey/goes16/CONUS/Channel01/20170821/GOES16_CONUS_20170821_020218_0.47_1km_33.3N_91.4W.nc4?dap4.ce=y#checksummode=ignore: NetCDF: libcurl failure
*** Fail: dap4://thredds-test.unidata.ucar.edu/thredds/dap4/casestudies/harvey/goes16/CONUS/Channel01/20170821/GOES16_CONUS_20170821_020218_0.47_1km_33.3N_91.4W.nc4?dap4.ce=y#checksummode=ignore

```
tests are trying to fetch things online, and it seems this was down now when i tried to build it.

**Comment by Micket** [comment]
> ugh
> 
> ```
> 206/227 Test #206: dap4_test_test_thredds ................***Failed  129.72 sec
> unknown argument test_thredds.sh
> test_thredds.sh:
> testing: dap4://thredds-test.unidata.ucar.edu/thredds/dap4/casestudies/harvey/goes16/CONUS/Channel01/20170821/GOES16_CONUS_20170821_020218_0.47_1km_33.3N_91.4W.nc4?dap4.ce=y#checksummode=ignore
> Error:curl error: Timeout was reached
> /dev/shm/netCDF/4.9.0/iimpi-2022a/easybuild_obj/ncdump/ncdump: dap4://thredds-test.unidata.ucar.edu/thredds/dap4/casestudies/harvey/goes16/CONUS/Channel01/20170821/GOES16_CONUS_20170821_020218_0.47_1km_33.3N_91.4W.nc4?dap4.ce=y#checksummode=ignore: NetCDF: libcurl failure
> *** Fail: dap4://thredds-test.unidata.ucar.edu/thredds/dap4/casestudies/harvey/goes16/CONUS/Channel01/20170821/GOES16_CONUS_20170821_020218_0.47_1km_33.3N_91.4W.nc4?dap4.ce=y#checksummode=ignore
> ```
> 
> tests are trying to fetch things online, and it seems this was down now when i tried to build it.
> 
> We can consider skipping these tests, if it's just dap4_ stuff that's affected here
> 
> ```python
> runtest = 'test ARGS="-E dap4"'
> ```
> 
> (of course, do the same for the gompi version)

**Comment by Micket** [comment]
Bah, sorry about that, i stumbled over the keyboard and accidentally closed

File: easybuild/easyconfigs/n/netCDF/netCDF-4.9.0-iimpi-2022a.eb
Comment (Micket) [inline easybuild/easyconfigs/n/netCDF/netCDF-4.9.0-iimpi-2022a.eb:42]:
```suggestion
    "-DENABLE_DAP_REMOTE_TESTS=OFF -DBUILD_SHARED_LIBS=OFF",
    "-DENABLE_DAP_REMOTE_TESTS=OFF -DBUILD_SHARED_LIBS=ON",
```

**Comment by Micket** [comment]
Sorry i was unclear, you don't need the -E dap4 thing thanks to the cmake configuration line that disables the exact tests instead.

I'm just holding off merging this because a big issue was discovered with HDF5, and we'll need to downgrade to 1.12.2 
See #16153

**Comment by SimonPinches** [comment]
@micket, Ah, okay.  I'll remove this last commit then.

File: easybuild/easyconfigs/n/netCDF/netCDF-4.9.0-iimpi-2022a.eb
Comment (Micket) [inline easybuild/easyconfigs/n/netCDF/netCDF-4.9.0-iimpi-2022a.eb:33]: So the HDF5 downgrade PR was finally merged now, so we can use 1.12 series here (turns out odd minor releases were just RCs, which we had overlooked):

```suggestion
    ('HDF5', '1.12.2'),
```

(I promise this is the last thing I'll request here :) )

**Review by Micket — DISMISSED** [review]

File: easybuild/easyconfigs/n/netCDF/netCDF-4.9.0-iimpi-2022a.eb
Comment (SebastianAchilles) [inline easybuild/easyconfigs/n/netCDF/netCDF-4.9.0-iimpi-2022a.eb:37]: As described in https://github.com/easybuilders/easybuild-easyconfigs/pull/16450 we need to add `bzip2` and `libxml2` as dep
```suggestion
    ('bzip2', '1.0.8'),
    ('libxml2', '2.9.13'),
]
```

**Review by SebastianAchilles — CHANGES_REQUESTED** [review]

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ generoso

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
zen2-ubuntu-2204 - Linux Ubuntu 22.04.1 LTS (Jammy Jellyfish), x86_64, AMD EPYC 7452 32-Core Processor (zen2), Python 3.10.6
See https://gist.github.com/8063812a14458d635284618526b39ee4 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
bwd-rockylinux-90 - Linux Rocky Linux 9.0 (Blue Onyx), x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), Python 3.9.10
See https://gist.github.com/4b022287009c54c17c4fd7507b04fe6b for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
skl-opensuse-154 - Linux openSUSE Leap 15.4, x86_64, Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz (skylake), Python 3.6.15
See https://gist.github.com/77417158b43ede130c2b5874583b3c6d for a full test report.

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen2

**Review by SebastianAchilles — APPROVED** [review]
lgtm

**Comment by SebastianAchilles** [comment]
Going in, thanks @SimonPinches!

**Pull request merged**
